### ISSUE TITLE
WoD 6.x.x Consecrated Letter - Brother Sammuel

### ISSUE THREAD

**Issue by Gecko60** [TrinityCore/TrinityCore#14213]
Warlords of Draenor
Core 6.0.3 (19342) Release X64

TrinityCore rev. 1cb513295bef 2015-02-19 17:12:18 +0100 (6.x branch) (Win64, Release)

You are given the quest "Consecrated Letter" as a paladin. However when you go to "Brother Sammuel" to turn it in, he does not speak to you when you click on him nothing happens. Therefore you are unable to proceed with your paladin class quest.

**Comment by Exodius** [comment]
```sql
-- Add Brother Sammuel as quest ender for Northshire quest "Consecrated Letter"
DELETE FROM `creature_queststarter` WHERE `quest`=3101;
INSERT INTO `creature_queststarter`(`id`,`quest`) VALUES (925,3101);

**Comment by Aokromes** [comment]
You wanted to write creature_questender :P not creature_queststarter

**Comment by tkrokli** [comment]
```sql
-- Add Brother Sammuel as quest ender for Northshire Abbey quest "Consecrated Letter"
DELETE FROM `creature_questender` WHERE `id`=925 AND `quest`=3101;
INSERT INTO `creature_questender`(`id`,`quest`) VALUES (925,3101);

**Comment by Gecko60** [comment]
I've implemented this into my test server, with support from @tkrokli however we have come to find out that the npc still does not talk to you, so you cannot turn in the quest, and also you cannot remove the quest from you quest log without using gm command .quest remove 3101

**Comment by Aokromes** [comment]
Do you realize that 6.x branch is heavy alpha and not intended for production enviroments? :)

**Comment by Gecko60** [comment]
Not sure what you mean by that, I'm just trying to fix things @Aokromes or make light of things that are not known or haven't been posted. Im here to learn, so I'm trying to work on the Northshire starter area. I figured this will be a good place for me to learn. :)

**Comment by Nyeriah** [comment]
He means abandoning quests and turning them in is probably yet not implemented at 6.x

**Comment by Gecko60** [comment]
Oh I see! :)

**Comment by Alastyr** [comment]
Maybe you should try to update the creature_template.

UPDATE creature_template SET npcflag=3 WHERE entry=925;

With this query you could speak with Brother Sammuel.

**Comment by tkrokli** [comment]
Ah, nice. Good call. Make sure the NPC is both quest giver and has gossip options.
http://collab.kpsn.org/display/tc/creature_template#creature_template-npcflag

So, are the above SQL snippets writeup material (after testing, ofc)?
```sql
-- Add Brother Sammuel as quest ender for Northshire Abbey quest "Consecrated Letter"
DELETE FROM `creature_questender` WHERE `id`=925 AND `quest`=3101;
INSERT INTO `creature_questender`(`id`,`quest`) VALUES (925,3101);

-- Update Brother Sammuel's questgiver and gossip status:
UPDATE creature_template SET npcflag=3 WHERE entry=925;

**Comment by Alastyr** [comment]
I have just taken the correct values from my 434 DB. In 434 this quest works.

**Comment by tkrokli** [comment]
OK, but what about source for those values in that 4.3.4 DB, is it from sniffs? :computer: Do you know if there is any issue or commit where those values were added? Would be cool to have it confirmed. :smirk_cat:

**Comment by Aokromes** [comment]
On 3.3.5a and 4.3.4 the npc flag for 925 is 51.

**Comment by Alastyr** [comment]
Yeah sorry. I search it in wrong DB. The correct flag are 51.

51 = 1 (gossip) + 2 (quest giver) + 16 (trainer) + 32 (class trainer).

Instead @tkrokli I can't find the commit or the issue, cause this NPC was added in patch 1.11. I think this flag is the same since Classic. 

Sorry for my mistakes and bad english.

UPDATE creature_template SET npcflag=51 WHERE entry=925;

**Comment by Aokromes** [comment]
BTW this mob and the database already have those values. this is a generic core bug.

**Comment by Alastyr** [comment]
Sorry then. I would not confuse. Just try to help. :)

**Comment by Alastyr** [comment]
This issue doesn't need "Branch-4.3.4" label.

**Comment by tkrokli** [comment]
@Gecko60 : Try running this SQL query on your test server and confirm if it works or not:
```sql
-- Add Brother Sammuel as quest ender for Northshire Abbey quest "Consecrated Letter"
DELETE FROM `creature_questender` WHERE `id`=925 AND `quest`=3101;
INSERT INTO `creature_questender`(`id`,`quest`) VALUES (925,3101);

-- Update Brother Sammuel's questgiver and gossip status:
UPDATE creature_template SET npcflag=51 WHERE entry=925;

**Comment by Gecko60** [comment]
I checked it the other day, it did not work.

**Comment by Gecko60** [comment]
Let me rephrase that. It went through just fine when I queried it, however it did nothing in game. Brother Sammuel still doesnt talk to you.

**Comment by Aokromes** [comment]
Like i said, it's a core bug.

**Comment by Nyeriah** [comment]
Maybe we should close this as the issue isn't the quest itself but something in the core which is still not running as it should? :)

**Comment by tkrokli** [comment]
Well, I second the motion if the issue is neither caused by anything in the quest or the NPC itself (and thus rendering the issue moot, since it needs to be fixed somewhere else. But do we have somewhere else to follow the progress? Like a milestone list or task list?

**Comment by Aokromes** [comment]
#14216
